### DISCUSSION TITLE
Getting empty list when accessing Repository List API

### DISCUSSION THREAD

**Discussion by togetherGithub** [community/community#23626] [New to GitHub]
<p>I am trying to get repository list of my account with the below curl command:<br>
<code>curl   -H "Authorization: some_random_token" https://api.github.com/users/myusername/repos</code></p>
<p>But getting an empty list from the server.</p>
<pre><code class="lang-auto">[
]
</code></pre>
<p><strong>Note :</strong> I have tried with different user with an associated token and that was successful. Anyone can help me with the issue.</p>

**Comment by jfullstackdev** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/8/88d401ef548446dfc0a64c8642e79384118e9c09.png" data-download-href="/uploads/short-url/jwr9VlqOU2E0vx1EYFNeyV5rnQZ.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181078341-f468012c-ba48-489e-aa2f-e6a42a7701c6.png" alt="image" data-base62-sha1="jwr9VlqOU2E0vx1EYFNeyV5rnQZ" width="690" height="180" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/8/88d401ef548446dfc0a64c8642e79384118e9c09_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">743×194 6.19 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>it’s empty to me also because you don’t have any public repo, but I cannot tell whether you have private repo/s</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/e/ae696b78ec3bcc114c87491a3295e1f2e94e2eeb.png" data-download-href="/uploads/short-url/oSUWkji0hdtdWx5E12p36ZsI9VV.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181078342-3b6a2f2d-5140-441c-a58d-05453c914a94.png" alt="image" data-base62-sha1="oSUWkji0hdtdWx5E12p36ZsI9VV" width="690" height="298" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/e/ae696b78ec3bcc114c87491a3295e1f2e94e2eeb_2_690x298.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/e/ae696b78ec3bcc114c87491a3295e1f2e94e2eeb_2_1035x447.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/a/e/ae696b78ec3bcc114c87491a3295e1f2e94e2eeb_2_1380x596.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/e/ae696b78ec3bcc114c87491a3295e1f2e94e2eeb_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1468×635 40 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>do you have any private repo? if you don’t have, then that will really be empty</p>

**Comment by togetherGithub** [comment]
<p>hi <a class="mention" href="/jdevstatic">@jdevstatic</a> , Thanks for your reply :smiling_face_with_three_hearts: . Yes I have private repositories and I did not actually noticed that the success result I was getting from another user was all public repositories :rofl: .  However Is there any way to list the private repositories  ?</p>

**Comment by jfullstackdev** [comment]
<p>try this one,</p>
<p><a href="https://docs.github.com/en/rest/reference/repos#list-repositories-for-the-authenticated-user" rel="noopener nofollow ugc">https://docs.github.com/en/rest/reference/repos#list-repositories-for-the-authenticated-user</a></p>
<p>you must be authenticated, so that you can see your private repos,</p>

**Comment by togetherGithub** [comment]
<p>Hi <a class="mention" href="/jdevstatic">@jdevstatic</a> , I have been trying in multiple way to get authenticated but still getting the same blank response. Don’t know what is wrong, can you please help?</p>
<pre><code class="lang-auto">curl  --header 'Accept: application/vnd.github.v3+json' \
--header 'Authorization: personal_access_token' \
https://api.github.com/users/togetherGithub/repos
</code></pre>
<pre><code class="lang-auto">curl  --header 'Accept: application/vnd.github.v3+json' \
--header 'Authorization: token personal_access_token' \
https://api.github.com/users/togetherGithub/repos
</code></pre>
<pre><code class="lang-auto">curl -u togetherGithub:personal_access_token \ 
https://api.github.com/users/togetherGithub/repos
</code></pre>

**Comment by jfullstackdev** [comment]
<p>it’s not those your presented, that’s for read only unauthenticated,</p>
<p>this one,<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/1/31ecf0055094bc6a3d4114660708a7bf966847ee.png" data-download-href="/uploads/short-url/77F20QupRPX7bmOZ9YYEb54iqE6.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181078347-5d247162-c8af-47ca-91e6-eccf15545425.png" alt="image" data-base62-sha1="77F20QupRPX7bmOZ9YYEb54iqE6" width="690" height="209" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/1/31ecf0055094bc6a3d4114660708a7bf966847ee_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">927×282 8.69 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jfullstackdev** [comment]
<p>without authentication, it will be<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/d/bd4cabcef431139399abc8638aca8df0a7ec450f.png" data-download-href="/uploads/short-url/r0CvyLVw089dTmh83ov8vvt3EbJ.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181078350-2b74e05e-8840-4fa1-8045-47ea53bdb8e1.png" alt="image" data-base62-sha1="r0CvyLVw089dTmh83ov8vvt3EbJ" width="690" height="136" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd4cabcef431139399abc8638aca8df0a7ec450f_2_690x136.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd4cabcef431139399abc8638aca8df0a7ec450f_2_1035x204.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/d/bd4cabcef431139399abc8638aca8df0a7ec450f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd4cabcef431139399abc8638aca8df0a7ec450f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1280×253 17.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by togetherGithub** [comment]
<p>Can you please share a curl command example with authentication ? So I can get the list of private repos ?</p>

**Comment by jfullstackdev** [comment]
<pre><code class="lang-auto">curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/user/repos
</code></pre>

**Comment by togetherGithub** [comment]
<p>OK. I think the authorization is ok now. I received result with the below command:</p>
<p><code>curl --header "Authorization: token Personal_Access_Token" https://api.github.com/repos/togetherGithub/A_Private_Repository_Name</code></p>
<p>But I still get empty result when execute the bellow command:</p>
<p><code>curl --header "Authorization: token Personal_Access_Token" https://api.github.com/users/togetherGithub/repos</code></p>

**Comment by jfullstackdev** [comment, marked as answer]
<aside class="quote no-group" data-username="togetherGithub" data-post="10" data-topic="218571">
<div class="title">
<div class="quote-controls"></div>
 togetherGithub:</div>
<blockquote>
<p><a href="https://api.github.com/users/togetherGithub/repos" rel="noopener nofollow ugc">https://api.github.com/users/togetherGithub/repos</a></p>
</blockquote>
</aside>
<p>that is different from this</p>
<p><code>https://api.github.com/user/repos</code></p>
<p>you simply add your token before this and call this command no need to change anything, see, it’s <code>user</code> not the generic <code>users</code>, that <code>user</code> is referring to you automatically, no need to include your name, because you are already authenticated</p>

**Comment by jfullstackdev** [comment]
<p>carefully look at this,<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/0/40976d318b5fd9a2f09201686dc017fe2089858c.png" data-download-href="/uploads/short-url/9doZfKyLLQdSK00oiP2pz7ckMSU.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181078353-38d5863e-5d13-42aa-928b-3b305e141b0e.png" alt="image" data-base62-sha1="9doZfKyLLQdSK00oiP2pz7ckMSU" width="690" height="327" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/0/40976d318b5fd9a2f09201686dc017fe2089858c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">894×424 27 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>it’s just <code>user</code> but authenticated,</p>
<p>the one that you are confused with the first one is this,<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/2/a22bca78da3662312c6c50f216525b70eac7b75a.png" data-download-href="/uploads/short-url/n8DaEo7pQV6KOkOurzfgdAWx0oO.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181078354-ef200638-d903-4d82-b5e9-702238c2e073.png" alt="image" data-base62-sha1="n8DaEo7pQV6KOkOurzfgdAWx0oO" width="593" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/2/a22bca78da3662312c6c50f216525b70eac7b75a_2_593x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/2/a22bca78da3662312c6c50f216525b70eac7b75a_2_889x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/a/2/a22bca78da3662312c6c50f216525b70eac7b75a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/2/a22bca78da3662312c6c50f216525b70eac7b75a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">929×782 81.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>it’s just getting a user, another user, in your case, your getting your own, that’s why it will not return your private repo, because you are using the API to get someone on GitHub, only the public information will be returned,</p>

**Comment by togetherGithub** [comment]
<p>Hi <a class="mention" href="/jdevstatic">@jdevstatic</a> Thanks for all the help :heart: . I finally was able to get the repository list by executing :partying_face: :<br>
curl --header “Authorization: token personal_access_token” <a href="https://api.github.com/user/repo" rel="noopener nofollow ugc">https://api.github.com/user/repo</a><br>
:cowboy_hat_face:<br>
Ref: <a href="https://github.community/t/getting-empty-list-when-accessing-repository-list-api/218571/11" class="inline-onebox">Getting empty list when accessing Repository List API - #11 by xdvrx1</a></p>
